examples/list_box_model/row_data/imp.rs

```diff
@@ -0,0 +1,85 @@
+use glib::subclass::prelude::*;
+use gtk::{glib, prelude::*};
+use std::cell::RefCell;
+
+// The actual data structure that stores our values. This is not accessible
+// directly from the outside.
+#[derive(Default)]
+pub struct RowData {
+    name: RefCell<Option<String>>,
+    count: RefCell<u32>,
+}
+
+// Basic declaration of our type for the GObject type system
+#[glib::object_subclass]
+impl ObjectSubclass for RowData {
+    const NAME: &'static str = "RowData";
+    type Type = super::RowData;
+    type ParentType = glib::Object;
+}
+
+// The ObjectImpl trait provides the setters/getters for GObject properties.
+// Here we need to provide the values that are internally stored back to the
+// caller, or store whatever new value the caller is providing.
+//
+// This maps between the GObject properties and our internal storage of the
+// corresponding values of the properties.
+impl ObjectImpl for RowData {
+    fn properties() -> &'static [glib::ParamSpec] {
+        use once_cell::sync::Lazy;
+        static PROPERTIES: Lazy<Vec<glib::ParamSpec>> = Lazy::new(|| {
+            vec![
+                glib::ParamSpecString::new(
+                    "name",
+                    "Name",
+                    "Name",
+                    None, // Default value
+                    glib::ParamFlags::READWRITE,
+                ),
+                glib::ParamSpecUInt::new(
+                    "count",
+                    "Count",
+                    "Count",
+                    0,
+                    100,
+                    0, // Allowed range and default value
+                    glib::ParamFlags::READWRITE,
+                ),
+            ]
+        });
+
+        PROPERTIES.as_ref()
+    }
+
+    fn set_property(
+        &self,
+        _obj: &Self::Type,
+        _id: usize,
+        value: &glib::Value,
+        pspec: &glib::ParamSpec,
+    ) {
+        match pspec.name() {
+            "name" => {
+                let name = value
+                    .get()
+                    .expect("type conformity checked by `Object::set_property`");
```

Comment:
I would just unwrap here,  we already have various examples & the book that explains that type conformity is checked by ObjectExt